RecoLocalCalo/HGCalRecProducers/interface/HGCalImagingAlgo.h

```diff
@@ -104,6 +104,7 @@ class HGCalImagingAlgo : public HGCalClusteringAlgoBase {
     iDesc.add<edm::ParameterSetDescription>("noises", descNestedNoises);
     edm::ParameterSetDescription descNestedNoiseMIP;
     descNestedNoiseMIP.add<bool>("scaleByDose", false);
+    descNestedNoiseMIP.add<double>("scaleByDoseFactor", 1.);
```

Comment:
Could you please remind me why all these parameters are defined inside a Reco producer, but they are only read and used in Simulation?

Reply:
Ciao @perrotta valid and good question.
The cluster algorithms import, by deliberate choice, parameters from the digitisation psets to access info like noise etc, and to ensure they use values consistent with those employed upstream. Together with parameters actually employed, the descriptor of the clustering producers end up having to include also  values which are not employed actually.
I don't particularly like this set up however I would not plan to re-engineer this aspect of the clustering and if these lines are not added redo crashes lamenting the producer is provided an invalid parameter.

I hope I am clear and accurate enough.

Reply:
I think the problem is not related to the underlying mechanism of validating the input parameters but to the way in which parameters used for reconstruction and simulation have been bundled together in the same PSet. Once you've made this choice, anything else is just a mere consequence of that.